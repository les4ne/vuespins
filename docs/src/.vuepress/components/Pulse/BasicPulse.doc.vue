<template>
  <div>
    <Pulse />
  </div>
</template>

<script>
import { Pulse } from 'vuespins'

export default {
  name: 'app',
  components: { Pulse }
}
</script>
