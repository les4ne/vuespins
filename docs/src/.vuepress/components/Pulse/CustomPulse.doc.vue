<template>
  <div>
    <Pulse color="#fb503b" size="80" speed="5" />
  </div>
</template>

<script>
import { Pulse } from 'vuespins'

export default {
  name: 'app',
  components: { Pulse }
}
</script>
