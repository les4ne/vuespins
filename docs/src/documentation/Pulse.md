# Pulse

## **Basic**

<br>
<client-only>
<Pulse />
</client-only>

### Props

Basic "Pulse" component

| property | propType | required | default | custom |
| :------: | :------: | :------: | :-----: | :----: |
| `color`  |  string  |  false   | #282c34 |   -    |
|  `size`  |  number  |  false   |   50    |   -    |
| `speed`  |  number  |  false   |   2.5   |   -    |

### Usage

<<< @/src/.vuepress/components/Pulse/BasicPulse.doc.vue{3,8,10-13}

## **Custom**

<br>
<client-only>
<Pulse color="#fb503b" :size="80" :speed="5" />
</client-only>

### Props

Custom "Pulse" component

| property | propType | required | default | custom  |
| :------: | :------: | :------: | :-----: | :-----: |
| `color`  |  string  |  false   | #282c34 | #fb503b |
|  `size`  |  number  |  false   |   50    |   80    |
| `speed`  |  number  |  false   |   2.5   |    5    |

### Usage

<<< @/src/.vuepress/components/Pulse/CustomPulse.doc.vue{3,8,10-13}
