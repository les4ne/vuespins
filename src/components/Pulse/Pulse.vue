<template>
  <div class="vs-pulse" :style="cssVars">
    <div class="double-bounce1"></div>
    <div class="double-bounce2"></div>
  </div>
</template>

<script>
export default {
  name: 'vs-pulse',
  props: {
    color: {
      type: String,
      default: '#282c34'
    },
    size: {
      type: Number,
      default: 50
    },
    speed: {
      type: Number,
      default: 2.5
    }
  },
  computed: {
    cssVars() {
      return {
        '--vs-pulse-size': this.size + 'px',
        '--vs-pulse-color': this.color,
        '--vs-pulse-speed': this.speed + 's',
        '--vs-pulse-delay': -this.speed + 1 + 's'
      }
    }
  }
}
</script>

<style scoped>
.vs-pulse {
  width: var(--vs-pulse-size);
  height: var(--vs-pulse-size);
  position: relative;
}

.double-bounce1,
.double-bounce2 {
  width: 100%;
  height: 100%;
  border-radius: 50%;
  background-color: var(--vs-pulse-color);
  opacity: 0.6;
  position: absolute;
  top: 0;
  left: 0;

  -webkit-animation: vs-bounce var(--vs-pulse-speed) infinite ease-in-out;
  animation: vs-bounce var(--vs-pulse-speed) infinite ease-in-out;
}

.double-bounce2 {
  -webkit-animation-delay: var(--vs-pulse-delay);
  animation-delay: var(--vs-pulse-delay);
}

@-webkit-keyframes vs-bounce {
  0%,
  100% {
    -webkit-transform: scale(0);
  }
  50% {
    -webkit-transform: scale(1);
  }
}

@keyframes vs-bounce {
  0%,
  100% {
    transform: scale(0);
    -webkit-transform: scale(0);
  }
  50% {
    transform: scale(1);
    -webkit-transform: scale(1);
  }
}
</style>
