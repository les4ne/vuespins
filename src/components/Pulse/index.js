import Pulse from './Pulse.vue'

export default Pulse
