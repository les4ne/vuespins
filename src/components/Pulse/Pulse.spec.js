import { shallowMount } from '@vue/test-utils'
import Pulse from './Pulse.vue'

describe('Pulse.vue', () => {
  it('renders correctly', () => {
    expect(shallowMount(Pulse).element).toMatchSnapshot()
  })
})
