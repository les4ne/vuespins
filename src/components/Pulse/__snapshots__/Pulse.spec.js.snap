// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pulse.vue renders correctly 1`] = `
<div
  class="vs-pulse"
  style="--vs-pulse-size: 50px; --vs-pulse-color: #282c34; --vs-pulse-speed: 2.5s; --vs-pulse-delay: -1.5s;"
>
  <div
    class="double-bounce1"
  />
   
  <div
    class="double-bounce2"
  />
</div>
`;
